nrnivmodl fails if neuron was installed using spack

trying to run `nrnivmodl` on a directory results in the following output:
```
cremones@daint102:ringtest (master) $ nrnivmodl mod
Creating x86_64 directory for .o files.

/users/cremones/test-use-cases/ringtest
ls: cannot access mod/*.inc: No such file or directory
mod/halfgap.mod
halfgap.mod
/users/cremones/software/spack-install/cray-CNL-haswell/intel-17.0.0.098/neuron-master-xefdyhbhchxb7vdt4pwowmbcozvmq77b/x86_64/bin/nocmodl halfgap
Translating halfgap.mod into halfgap.c
Thread Safe
/users/cremones/software/spack/lib/spack/env/intel/icc -DHAVE_CONFIG_H -I. -I.. -I/users/cremones/software/spack-install/cray-CNL-haswell/intel-17.0.0.098/neuron-master-xefdyhbhchxb7vdt4pwowmbcozvmq77b/include/nrn -I/users/cremones/software/spack-install/cray-CNL-haswell/intel-17.0.0.098/neuron-master-xefdyhbhchxb7vdt4pwowmbcozvmq77b/x86_64/lib     -O2 -g  -c halfgap.c
Spack compiler must be run from Spack! Input 'SPACK_PREFIX' is missing.
/users/cremones/software/spack-install/cray-CNL-haswell/intel-17.0.0.098/neuron-master-xefdyhbhchxb7vdt4pwowmbcozvmq77b/x86_64/bin/nrniv_makefile:101: recipe for target 'halfgap.o' failed
make: *** [halfgap.o] Error 1
```

this seems to happen because in the file
`/users/cremones/software/spack-install/cray-CNL-haswell/intel-17.0.0.098/neuron-master-xefdyhbhchxb7vdt4pwowmbcozvmq77b/x86_64/bin/nrniv_makefile` 
is hard-coded the path to the compiler, i.e. at line 47:
`CC = /users/cremones/software/spack/lib/spack/env/intel/icc`

is there a way around this?

--- a/packages/neuron/package.py
+++ b/packages/neuron/package.py
@@ -294,6 +294,22 @@ class Neuron(Package):
             make('install')
             self.profiling_wrapper_off()
 
+    @run_after('install')
+    def filter_compilers(self):
+        """run after install to avoid spack compiler wrappers
+        getting embded into nrnivmodl script"""
+
+        arch = self.get_neuron_arch_dir()
+        nrnmakefile = join_path(self.prefix, arch, 'bin/nrniv_makefile')
+
+        kwargs = {
+            'backup': False,
+            'string': True
+        }
+
+        filter_file(env['CC'],  self.compiler.cc, nrnmakefile, **kwargs)
+        filter_file(env['CXX'], self.compiler.cxx, nrnmakefile, **kwargs)
+
     def setup_environment(self, spack_env, run_env):
         arch = self.get_neuron_arch_dir()
         run_env.prepend_path('PATH', join_path(self.prefix, arch, 'bin'))

